tests/socket.rs

```diff
@@ -1341,3 +1341,25 @@ fn original_dst_ipv6() {
         Err(err) => assert_eq!(err.raw_os_error(), Some(libc::EOPNOTSUPP)),
     }
 }
+
+#[test]
+#[cfg(all(feature = "all", any(target_os = "freebsd", target_os = "linux")))]
+fn tcp_congestion() {
+    let socket: Socket = Socket::new(Domain::IPV4, Type::STREAM, None).unwrap();
+    // Get and set current tcp_ca
+    let origin_tcp_ca = socket
+        .tcp_congestion()
+        .expect("failed to get tcp congestion algorithm");
+    socket
+        .set_tcp_congestion(&origin_tcp_ca)
+        .expect("failed to set tcp congestion algorithm");
+    // Return a Err when set a non-exist tcp_ca
+    socket
+        .set_tcp_congestion(b"tcp_congestion_does_not_exist")
+        .unwrap_err();
```

Comment:
Can you also try setting a different algorithm. I think "cubic" and "reno" are pretty common, you can try those two.

Reply:
I will try to set them. The original reason I do not explicitly set a algorithm is that the default ca for FreeBSD is newreno, not sure if others are available. But let's give it a try on CI. If not, I think it is better to split the test for different os if you agree.

Reply:
According to the documentation of FreeBSD, it only supports cubic not reno(since it uses newreno). Do we only set "cubic" in the test or split the test according to os?

Reply:
Ok, I will implement like this.